pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -1152,7 +1150,7 @@ func TestValidatePipelineResults_Failure(t *testing.T) {
 		},
 	}}
 	for _, tt := range tests {
-		err := validatePipelineResults(tt.results)
+		err := validatePipelineResults(tt.results, []PipelineTask{})
```

Comment:
just so you don't have to modify that test case - that it stays focused on the result expression - you can pass in the pipelinetask name a-task:

```suggestion
		err := validatePipelineResults(tt.results, []PipelineTask{{Name: "a-task"})
```

Reply:
and add your own test cases to `TestValidatePipelineResults_Failure` given that you've changed the logic for `validatePipelineResults`